interface/patient_file/encounter/find_code_history.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * Lookup past and current dx codes (favorites)
+ * (Temporary rest test interface until add a model)
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2019 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once("../../globals.php");
+require_once($GLOBALS["srcdir"] . "/options.inc.php");
+
+use OpenEMR\Core\Header;
+
+function unique_array_by_key($array, $key)
+{
+    $temp_array = array();
+    $i = 0;
+    $key_array = array();
+
+    foreach ($array as $val) {
+        if (!in_array($val[$key], $key_array)) {
+            $key_array[$i] = $val[$key];
+            $temp_array[$i] = $val;
+        }
+        $i++;
+    }
+    return $temp_array;
+}
+
+function get_history_codes($pid)
+{
+    $origin = xlt('Patient Problems');
+    $probcodes = array();
+    $bld = '';
+    $dres = sqlStatementNoLog(
+        "SELECT diagnosis as codes FROM lists " .
+        "Where activity = 1 And type = ? And pid = ?",
+        array('medical_problem', $pid)
+    );
+    while ($diag = sqlFetchArray($dres)) {
+        if (strpos($diag['codes'], 'ICD') === false) {
```

Comment:
Is the goal here to only show ICD codes or any codes that are being used as diagnosis codes (for example, in another country that uses snomed for dx codes). If goal is to show diagnosis codes, then can use the collect_codetypes() function in custom/code_types.inc.php . Check out interface/patient_file/summary/add_edit_issue.php for example of use.

Reply:
btw, this is collected from the settings stored in Administration->Lists->Code Types

Reply:
yep, true. I'm too fixed on lab's. Only place I care about code type. thanks